qt/widgets/common/src/ImageInfoWidget.cpp

```diff
@@ -44,17 +44,29 @@ void ImageInfoWidget::cursorAt(const double x, const double y, const double sign
 void ImageInfoWidget::showInfo(const ImageInfoModel::ImageInfo &info) {
   if (info.empty())
     return;
+  int signalIndex = -1;
 
   const auto itemCount(info.size());
   setColumnCount(itemCount);
   for (int i = 0; i < itemCount; ++i) {
-    auto header = new QTableWidgetItem(info.name(i));
+    const auto name = info.name(i);
+    if (name == "Signal") {
+      signalIndex = i;
+    }
+    auto header = new QTableWidgetItem(name);
     header->setFlags(header->flags() & ~Qt::ItemIsEditable);
     setItem(0, i, header);
     auto value = new QTableWidgetItem(info.value(i));
     value->setFlags(header->flags() & ~Qt::ItemIsEditable);
     setItem(1, i, value);
   }
+  if (signalIndex >= 0) {
+    if (!m_showSignal) {
+      hideColumn(signalIndex);
+    } else {
+      showColumn(signalIndex);
+    }
+  }
```

Comment:
I'm a bit hesitant about this logic being placed in the view. Could we have the GUI logic and `m_showSignal` in the presenter and just do a call to a presenter method in the view (similar to how `setWorkspace()` is done).

Reply:
Just been doing this, got everythin in place and the realise that `m_view` in the presenter is an `IImageInfoWidget` (the interface class) so you can't call any of the `QTableWidget` methods on it. What do you think is the best approach in that case?

Reply:
`IImageInfoWidget` already has a `setRowCount()` method that gets overridden with a blank method in `IImageInfoWidgetMini.cpp`. I think the best approach would be to add a couple of new methods to the interface & views that can be called to show/hide columns.